Source/Lib/Codec/EbEntropyCoding.c

```diff
@@ -4968,70 +4968,37 @@ static void CodeProfileTier(
         // "general_max_12bit_constraint_flag"
         WriteFlagCavlc(
            bitstreamPtr,
-           1);
+           (scsPtr->encoderBitDepth <= EB_12BIT));
 
         // "general_max_10bit_constraint_flag"
-        if(scsPtr->encoderBitDepth <= EB_10BIT || scsPtr->staticConfig.constrainedIntra == EB_TRUE)
-        {
-            WriteFlagCavlc(
-               bitstreamPtr,
-               1);
-        } else
-        {
-            WriteFlagCavlc(
-               bitstreamPtr,
-               0);
-        }
+        WriteFlagCavlc(
+           bitstreamPtr,
+           (scsPtr->encoderBitDepth <= EB_10BIT));
 
         // "general_max_8bit_constraint_flag"
-        //if(scsPtr->encoderBitDepth == EB_8BIT)
-        if(scsPtr->encoderBitDepth == EB_8BIT && (scsPtr->chromaFormatIdc == EB_YUV444 || scsPtr->staticConfig.constrainedIntra == EB_TRUE))
-        {
-            WriteFlagCavlc(
-               bitstreamPtr,
-               1);
-        } else
-        {
-            WriteFlagCavlc(
-               bitstreamPtr,
-               0);
-        }
+        WriteFlagCavlc(
+           bitstreamPtr,
+           (scsPtr->encoderBitDepth <= EB_8BIT && scsPtr->chromaFormatIdc != EB_YUV422));
 
         // "general_max_422chroma_constraint_flag"
-        if(scsPtr->chromaFormatIdc == EB_YUV422 || (scsPtr->chromaFormatIdc == EB_YUV420 && scsPtr->staticConfig.constrainedIntra == EB_TRUE))
-        {
-            WriteFlagCavlc(
-               bitstreamPtr,
-               1);
-        } else
-        {
-            WriteFlagCavlc(
-               bitstreamPtr,
-               0);
-        }
+        WriteFlagCavlc(
+           bitstreamPtr,
+           (scsPtr->chromaFormatIdc <= EB_YUV422));
 
         // "general_max_420chroma_constraint_flag"
-        if(scsPtr->chromaFormatIdc == EB_YUV420)
-        {
-            WriteFlagCavlc(
-               bitstreamPtr,
-               1);
-        } else
-        {
-            WriteFlagCavlc(
-               bitstreamPtr,
-               0);
-        }
+        WriteFlagCavlc(
+           bitstreamPtr,
+           (scsPtr->chromaFormatIdc <= EB_YUV420));
 
         // "general_max_monochrome_constraint_flag"
         WriteFlagCavlc(
            bitstreamPtr,
-           0);
+           (scsPtr->chromaFormatIdc <= EB_YUV400));
 
         // "general_intra_constraint_flag"
         WriteFlagCavlc(
            bitstreamPtr,
-           (scsPtr->staticConfig.constrainedIntra == EB_TRUE));
+           (scsPtr->intraPeriodLength == 0));
```

Comment:
I didn't find any related doc. Could you let me know how general_intra_constraint_flag is set? why set when gop is 1 instead of from staticConfig.constrainedIntra? Thank you.

Reply:
@tianjunwork I didn't find it too in ITU-T doc. But HM or x265 set this flag only when only using intra-picture prediction mode. 
(https://hevc.hhi.fraunhofer.de/HM-doc/_t_app_enc_cfg_8cpp_source.html : line 2253, https://github.com/videolan/x265/blob/master/source/encoder/level.cpp#L67)

Reply:
Thanks pointing out the reference code.

Reply:
According to "Table A.2 – Bitstream indications for conformance to format range extensions profiles" of HEVC spec, general_intra_constraint_flag should be set to 1 for intra (prediction) profiles or Still Picture ones. So this change is correct.